tests/Service/EnterpriseSearch/EnterpriseSearchServiceTest.php

```diff
@@ -1243,36 +1248,60 @@ public function testAddDocumentsEmpty(): void
         $this->assertEqualsCanonicalizing([], $resultIds);
     }
 
-    public function testAddDocumentsError(): void
+
+    /**
+     * Test that when an error occurs, the remaining documents are processed and errors are logged.
+     */
+    public function testAddDocumentsWithErrorAndContinue(): void
     {
         $documentOne = $this->objFromFixture(DataObjectFake::class, 'one');
-        $documents = [DataObjectDocument::create($documentOne)];
+        $documentTwo = $this->objFromFixture(DataObjectFake::class, 'two');
+        $documentThree = $this->objFromFixture(DataObjectFake::class, 'three');
 
-        $this->expectExceptionMessage('Testing failure');
```

Comment:
Change in behaviour - rather than throw an exception, we now log the error and continue.